lib/sproxyd.js

```diff
@@ -187,9 +187,21 @@ class SproxydClient {
                 log.debug('stored to sproxyd', { newKey });
                 return callback(null, newKey);
             });
+            request.on('finish',
+                () => log.debug('finished sending PUT chunks to sproxyd', {
+                    component: 'sproxydclient',
+                    method: '_handleRequest',
+                    contentLength: size,
+                }));
             stream.pipe(request);
-            request.on('end', () => request.end);
-            log.debug('finished sending PUT chunks to sproxyd');
+            stream.on('error', err => {
+                log.error('error from readable stream', {
+                    error: err,
+                    method: '_handleRequest',
+                    component: 'sproxydclient',
+                });
+                request.end();
```

Comment:
Yes, `request.end()` should close the connection to remote host. If not for this change, then it would behave like you said "kees the connection open with the remote host closing the connection after timeout"